app/mailers/contact_mailer.rb

```diff
@@ -1,13 +1,37 @@
 # encoding: UTF-8
+require 'digest/sha2'
 class ContactMailer < ActionMailer::Base
   default from: 'Spindelman <svarainte@fsektionen.se>', parts_order: ["text/plain", "text/html"]
   default subject: 'Meddelande via Fsektionen.se'
 
-  def contact_email(name, email, msg, kontakt, copy)
-    @name, @email, @msg, @kontakt = name, email, msg, kontakt
-    if @name.present? && @email.present? && @msg.present? && kontakt.present?
-      recipients = copy == '1' ? [@kontakt.email, email] : @kontakt.email
-      mail from: %(#{@name}   <#{@email}>), to: recipients
+  def mail(contact)
+    set_message_id
+    @contact = contact
+
+    if @contact.present?
+      mail(to: @contact.recipients,
+           reply_to: recipient(email: @contact.send_email, name: @contact.send_name),
+           cc: copy(copy: @contact.copy, email: @contact.send_email))
+    end
+  end
+
+  private
+
+  def recipient(name: nil, email:)
+    if name.present?
+      "#{name} <#{email}>"
+    else
+      send_email
     end
   end
+
+  def copy(copy: false, email: nil)
+    if copy
+      send_email
+    end
+  end
+
+  def set_message_id(sent_at: Time.zone.now)
+    headers["Message-ID"] = "<#{Digest::SHA2.hexdigest(sent_at.to_i.to_s)}@fsektionen.se>"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.